controller/cache/cache.go

```diff
@@ -191,9 +192,12 @@ func (c *liveStateCache) Run(ctx context.Context) {
 			// cluster resources must be watched only if cluster has at least one app
 			if (event.Type == watch.Deleted || !hasApps) && ok {
 				info.cancel()
+				watchingClustersLock.Lock()
 				delete(watchingClusters, event.Cluster.Server)
+				watchingClustersLock.Unlock()
```

Comment:
should this be on defer?

Reply:
to use defer I would have to create one line function, so I decided to just execute unlock manually.